src/algorand_transactions/transaction.rs

```diff
@@ -301,10 +301,20 @@ impl AlgorandTransaction {
             .map(|ref msg_pack_bytes| Self::prefix_tx_byte(msg_pack_bytes))
     }
 
+    pub fn assign_group_id(&self, group_id: AlgorandHash) -> Self {
+        let mut mutable_self = self.clone();
+        mutable_self.group = Some(group_id);
+        mutable_self
+    }
+
     pub fn to_raw_tx_id(&self) -> Result<AlgorandHash> {
         AlgorandHash::from_slice(&sha512_256_hash_bytes(&self.encode_for_signing()?))
     }
 
+    pub fn group(&self) -> Option<AlgorandHash> {
+        self.group
+    }
+
```

Comment:
Might be better here to make a more useful helper, since `group` is already `pub` in this struct, so a user can just get this exact result via `tx.group` instead of using this fxn `tx.group()`. So it's redundant. 

Instead, let's return a result, so that we can help the user get the underlying group hash:

```suggestion
    pub fn group(&self) -> Result<AlgorandHash> {
        match self.group {
            Some(hash) => Ok(*hash),
            None => Err("No group ID set in transaction!".into()),
        }
    }

```